File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
import java.lang.Math;
import java.util.stream.IntStream;

class ArmstrongNumbers {
	boolean isArmstrongNumber(int numberToCheck) {
		String s = Integer.toString(numberToCheck);
		int length = s.length();
		return IntStream.range(0, length)
			.reduce(0, (a, b) -> a += Math.pow((int)(s.charAt(b) - '0'), length))
			== numberToCheck;
	}
}

File: cpp/armstrong-numbers/armstrong_numbers.cpp
#include "armstrong_numbers.h"

#include <vector>
#include <cmath>

bool armstrong_numbers::is_armstrong_number(int n) {
  std::vector<int> digits;
  int temp = n;
  int length = 0;
  while (temp > 0) {
    digits.push_back(temp % 10);
    temp /= 10;
    length++;
  }
  int sum = 0;
  for (int digit : digits) {
    sum += pow(digit, length);
  }
  return sum == n;
}